File: problems/basic/191/solution.py
# Accepted: 29ms (96.20%), 17.48MB (8.69%)

class Solution:
    def hammingWeight(self, n: int) -> int:
        dist = 0
        while n != 0:
            if n & 1:
                dist += 1
            n >>= 1
        return dist

File: problems/basic/191/solution.ts
// Accepted: 59ms (66.61%), 44.18MB (90.86%)

function hammingWeight(n: number): number {
    let dist = 0;
    while (n !== 0) {
        if ((n & 1) == 1) {
            dist++;
        }
        n >>>= 1;
        // note that the >> operator is signed shift (e.g., -1 >> 1 === -1)
    }
    return dist;
};
